lang/ghc/Portfile

```diff
@@ -134,14 +133,18 @@ compiler.library_path \
 # binaries into ${prefix}, not DESTDIR.  Work around this by setting
 # --prefix=${destroot}${prefix} and not setting DESTDIR
 
-if {![variant_isset "bootstrap"]} {
+if { [variant_isset "nobootstrap"] } {
     set bootstrap_dir ${destroot}${prefix}
 } else {
     set bootstrap_dir ${workpath}/bootstrap
 
     pre-configure {
         xinstall -m 0755 -d ${bootstrap_dir}
     }
+
+    configure.env-append \
+        PATH=$env(PATH):${bootstrap_dir}/bin \
+        SPHINXBUILD='${prefix}/bin/sphinx-build-${python3_version}'
```

Comment:
You seem to have two separate ‘not nobootstrap’ defined, e.g. the one here and the one starting line 47. These can and should be merged together.

Reply:
Not, partly why I noticed this is you only define the variable python3_version in the first block, but use it in both. I am not sure if this is robust or not, or even valid.

Reply:
I'll do what you say, but maybe with some protest.

The first block handles "mirror this stuff" things.

The second block handles "build this stuff" things, which comes after a bunch of variant-independent "build this stuff" directives.

Logically, that makes sense as one works through each step in the build process.

Also, won't making those edits break PR https://github.com/macports/macports-ports/pull/5124, which isn't yet merged? (hint hint)

Reply:
> Note, partly why I noticed this is you only define the variable python3_version in the first block, but use it in both. I am not sure if this is robust or not, or even valid.

Why wouldn't it be valid? They're just `if` blocks. The variable is defined in the first, and used in the second. That will always work, right?

Reply:
@cjones051073 
> You seem to have two separate ‘not nobootstrap’ variant blocks defined, e.g. the one here and the one starting line 47. These can and should be merged together.

Making sure that we’re on the same page. Those are `if` blocks, not `variant` blocks, and there are good reasons for structuring it the way it is. Would you please clarify?

Reply:
I don’t know either tcl to be sure, but in many other languages it would not work. The variable define in the first scope would only be valid in that scope, not the second. 

That aside, I just think it obfuscating a bit to have two blocks for the same if clause.

Reply:
> I'll do what you say, but maybe with some protest.
> 
> The first block handles "mirror this stuff" things.
> 
> The second block handles "build this stuff" things, which comes after a bunch of variant-independent "build this stuff" directives.
> 
> Logically, that makes sense as one works through each step in the build process.
> 
> Also, won't making those edits break PR #5124, which isn't yet merged? (hint hint)

Be careful. Things do not necessary happen in a port file in the exact order they are written in the portfile.

Reply:
Got it. Thanks for the comments. I'll refactor and corporate your merge with a rebase.